tests/unit/states/test_vmc_nat_rules.py

```diff
@@ -0,0 +1,711 @@
+        "parent_path": "/infra/tier-1s/cgw/nat/USER",
+        "unique_id": "7039a170-6f18-43b0-9ae0-a172c48fae8a",
+        "marked_for_delete": False,
+        "overridden": False,
+        "_create_time": 1617867323573,
+        "_create_user": "pnaval@vmware.com",
+        "_last_modified_time": 1617867323577,
+        "_last_modified_user": "pnaval@vmware.com",
+        "_system_owned": False,
+        "_protection": "NOT_PROTECTED",
+        "_revision": 0,
+    }
+
+    return response
+
+
+@pytest.fixture
+def mocked_error_response():
+    error_response = {
+        "error": "The credentials were incorrect or the account specified has been locked."
+    }
+    return error_response
+
+
+def test_present_state_when_error_from_get_by_id(mocked_ok_response, mocked_error_response):
+    mock_get_by_id = create_autospec(
+        vmc_nat_rules_exec.get_by_id, return_value=mocked_error_response
+    )
+
+    with patch.dict(vmc_nat_rules.__salt__, {"vmc_nat_rules.get_by_id": mock_get_by_id}):
+        result = vmc_nat_rules.present(
+            name="test_present",
+            hostname="hostname",
+            refresh_key="refresh_key",
+            authorization_host="authorization_host",
+            org_id="org_id",
+            sddc_id="sddc_id",
+            tier1="tier1",
+            nat="nat",
+            nat_rule=mocked_ok_response["id"],
+        )
+
+    assert result is not None
+    assert result["changes"] == {}
+    assert (
+        result["comment"]
+        == "The credentials were incorrect or the account specified has been locked."
+    )
+    assert not result["result"]
+
+
+def test_present_state_when_error_from_create(mocked_error_response):
+    mock_get_by_id = create_autospec(vmc_nat_rules_exec.get_by_id, return_value={})
+    mock_create = create_autospec(vmc_nat_rules_exec.create, return_value=mocked_error_response)
+
+    with patch.dict(
+        vmc_nat_rules.__salt__,
+        {
+            "vmc_nat_rules.get_by_id": mock_get_by_id,
+            "vmc_nat_rules.create": mock_create,
+        },
+    ):
+        result = vmc_nat_rules.present(
+            name="test_present",
+            hostname="hostname",
+            refresh_key="refresh_key",
+            authorization_host="authorization_host",
+            org_id="org_id",
+            sddc_id="sddc_id",
+            tier1="tier1",
+            nat="nat",
+            nat_rule="nat_rule",
+        )
+
+    assert result is not None
+    assert result["changes"] == {}
+    assert (
+        result["comment"]
+        == "The credentials were incorrect or the account specified has been locked."
+    )
+    assert not result["result"]
+
+
+def test_present_state_when_error_from_update(mocked_ok_response, mocked_error_response):
+    mock_get_by_id = create_autospec(vmc_nat_rules_exec.get_by_id, return_value=mocked_ok_response)
+    mock_update = create_autospec(vmc_nat_rules_exec.update, return_value=mocked_error_response)
```

Comment:
**suggestion** I'm not sure which PR it was but as @dhiltonp suggested, it would be more appropriate to push these mocks into the fixtures. If you need help - attend one of the [Test Clinics on Twitch](https://www.twitch.tv/saltstackinc/schedule).

Reply:
here, we have seggregated the fixtures mocked_ok_response and mocked_error_repsonse as per @dhiltonp's suggestion